### PR TITLE
Fix invalid GeoJSON returned

### LINKED ISSUE #7627: Multipolygon shape cause invalid GeoJSON
## Description

An unbalanced multipolygon shape will generate an invalid GeoJSON with null coordinates

## Steps to reproduce

Create an unbalanced Multipolygon shape and call toGeoJson:

```javascript
const poly = L.polygon([[[
		[51.509, -0.08],
		[51.503, -0.06],
		[51.51, -0.046]
	],
        [
		[51.508, -0.06],
		[51.504, -0.06],
		[51.509, -0.05]
	]
     ], [
           [51.499, -0.08],
	   [51.493, -0.06],
	   [51.48, -0.046]
        ]])

console.log(poly.toGeoJSON().geometry.coordinates)
``` 
Returns:
```json
[
    [
        [
            [
                -0.08,
                51.509
            ],
            [
                -0.06,
                51.503
            ],
            [
                -0.046,
                51.51
            ],
            [
                -0.08,
                51.509
            ]
        ],
        [
            [
                -0.06,
                51.508
            ],
            [
                -0.06,
                51.504
            ],
            [
                -0.05,
                51.509
            ],
            [
                -0.06,
                51.508
            ]
        ]
    ],
    [
        [
            null
        ],
        [
            null
        ],
        [
            null
        ]
    ]
]
```

## Environment
Leaflet version : 1.7.1

## Minimal example reproducing the issue

Please see https://plnkr.co/edit/h7neyNZ5M8sLId7s and click on the polygon

### PR DESCRIPTION
**Pull request by steff1986**
Root cause is issue in argument 'levelsDeep' if unbalanced arrays are inputted. Then, the input argument may not represent the actual dimension of the input array. A check for flatness corrects the issue though there might be better solutions.
Fix #7627.

### CODE REVIEW

File: src/layer/GeoJSON.js [hunk 1]
```diff
@@ -265,12 +265,13 @@ export function latLngToCoords(latlng, precision) {
 // @function latLngsToCoords(latlngs: Array, levelsDeep?: Number, closed?: Boolean): Array
 // Reverse of [`coordsToLatLngs`](#geojson-coordstolatlngs)
 // `closed` determines whether the first point should be appended to the end of the array to close the feature, only used when `levelsDeep` is 0. False by default.
+// Check for flat arrays required to ensure unbalanced arrays are correctly converted in recursion
```
Comment (mourner) [inline src/layer/GeoJSON.js:268]: Nit: I'd move the comment inside so that it's not exposed as public documentation.

**Review by mourner — CHANGES_REQUESTED** [review]
Can you please provide a test case that demonstrates why the change is needed?

**Comment by Falke-Design** [comment]
Linking issue:
Fix: #7627

**Review by johnd0e — CHANGES_REQUESTED** [review]
Please resolve conflicts

**Review by mourner — APPROVED** [review]

**Pull request merged**